File: javascript/high-scores/high-scores.js
//
// This is only a SKELETON file for the 'High Scores' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class HighScores {
  constructor(input) {
    this._scores = input
  }

  get scores() {
    return this._scores
  }

  get latest() {
    return this._scores[this._scores.length - 1]
  }

  get personalBest() {
    return Math.max(...this._scores)
  }

  get personalTopThree() {

    let topThree = []
    let copyOfScores = [...this._scores]
    let maxValue = 0
    let amountOfValues = copyOfScores.length > 2 ? 3 : copyOfScores.length

    for (let i = 0; i < amountOfValues; i++) {

      maxValue = Math.max(...copyOfScores)
      topThree.push(maxValue)
      copyOfScores.splice(copyOfScores.indexOf(maxValue), 1)

    }

    return topThree
  }
}

File: python/high-scores/high_scores.py
def latest(scores):
    return scores[len(scores) - 1]

def personal_best(scores):
    best_score = 0

    for score in scores:
        if score > best_score:
            best_score = score

    return best_score 

def personal_top_three(scores):
    list_top = []
    num_of_item = 0

    if len(scores) > 2:
        num_of_item = 3
    else:
        num_of_item = len(scores)

    scores.sort(reverse = True)
    for x in range(0, num_of_item):
        list_top.append(scores[x])
    
    return list_top

